app/src/main/java/org/mozilla/focus/autocomplete/AutocompleteAddFragment.kt

```diff
@@ -60,10 +60,17 @@ class AutocompleteAddFragment : Fragment() {
                     .toLowerCase()
                     .removePrefixesIgnoreCase("http://", "https://", "www.")
 
-            if (domain.isEmpty()) {
-                domainView.error = getString(R.string.preference_autocomplete_add_error)
-            } else {
-                saveDomainAndClose(activity.applicationContext, domain)
+            launch(UI) {
+                when {
+                    domain.isEmpty()
+                    -> domainView.error = getString(R.string.preference_autocomplete_add_error)
```

Comment:
I don't think this is correct style. You want to keep the arrows on the same line like:
```kotlin
domain.isEmpty() -> {
    // Multiline things here
}
```